### PR TITLE
Renderable locks

### PR DESCRIPTION
**Pull request by vlovich**
This lets you use a @renderer wrapper to generate the locks to acquire.

### CODE REVIEW

File: master/docs/manual/cfg-buildsteps.rst [hunk 1]
```diff
@@ -144,6 +144,8 @@ Arguments common to all :class:`BuildStep` subclasses:
 
 ``locks``
     a list of ``Locks`` (instances of :class:`buildbot.locks.SlaveLock` or :class:`buildbot.locks.MasterLock`) that should be acquired before starting this :class:`Step`.
+    Alternatively this could be a renderable that returns this list during build execution. This lets you defer picking the locks to acquire until the build
```
Comment (sa2ajj) [inline master/docs/manual/cfg-buildsteps.rst]: One sentence per line!

**Comment by tardyp** [comment]
:+1: I would prefer this happens on master first

**Comment by vlovich** [comment]
Ported to master: https://github.com/buildbot/buildbot/pull/1670

**Comment by tardyp** [comment]
👍

Le dim. 17 mai 2015 11:10, Vitali Lovich <notifications@github.com> a
écrit :

**Comment by yetanotherion** [comment]
:+1:

**Pull request merged**